x/leverage/keeper/keeper.go

```diff
@@ -377,7 +370,42 @@ func (k Keeper) Liquidate(
 
 // FastLiquidate
 func (k Keeper) FastLiquidate(
-	_ sdk.Context, _, _ sdk.AccAddress, _, _ string,
+	ctx sdk.Context, liquidatorAddr, borrowerAddr sdk.AccAddress, repayDenom, rewardDenom string,
 ) (repaid sdk.Coin, reward sdk.Coin, err error) {
-	return sdk.Coin{}, sdk.Coin{}, err
+	if err := k.validateAcceptedDenom(ctx, repayDenom); err != nil {
+		return sdk.Coin{}, sdk.Coin{}, err
+	}
+	if err := k.validateAcceptedDenom(ctx, rewardDenom); err != nil {
+		return sdk.Coin{}, sdk.Coin{}, err
+	}
+	uRewardDenom := types.ToUTokenDenom(rewardDenom)
+
+	tokenRepay, uTokenReward, _, err := k.getLiquidationAmounts(
```

Comment:
nope, that would complicate things and might lead to unintended outcomes (especially since SupplyCollateral is the default on frontend - that means a non-collateral position was likely intentional by an advanced user and we shouldn't automatically collateralize)